module/fastqc.nf

```diff
@@ -0,0 +1,43 @@
+/*
+*   Nextflow module for running FASTQC
+*
+*   @input fq_path path path to the input FASTQ file
+*   @output fastqc_output_dir dir unzipped FASTQC output directory
+*/
+
+include { generate_standard_filename } from '../external/pipeline-Nextflow-module/modules/common/generate_standardized_filename/main.nf'
+
+process assess_ReadQuality_FastQC {
+    container params.docker_image_fastqc
+
+    publishDir path: "${params.workflow_output_dir}/output",
+        pattern: "${output_filename}",
+        mode: "copy",
+        enabled: true
+    ext log_dir_suffix: { "-${id}" }
+
+    input:
+        tuple val(orig_id), val(id), path(path), val(read_length), val(sample_type)
+
+    output:
+        path("${output_filename}")
+
+    script:
+    output_filename = generate_standard_filename("FastQC-${params.fastqc_version}",
+        params.dataset_id,
+        id,
+        [:])
+
+    """
+    set -euo pipefail
+    mkdir "${output_filename}"
+    fastqc \
+        --outdir "${output_filename}" \
```

Comment:
- In this case I used `generate_standard_filename` to create an additional output directory with the `dataset_id` and the `id` extracted from the input BAM files.  
- An alternate possibility is to output to "./" and then rename the files within `${bam_prefix}_fastqc` (i.e. the FastQC directory that is always created) to include the standard_filename components.